app/AppComponents.scala

```diff
@@ -53,13 +54,20 @@ class AppComponents(context: Context, identity: AppIdentity, creds: AWSCredentia
   val settingsFile = identity match {
     case identity: AwsIdentity if identity.stage == "PROD" => "gutools.co.uk.settings.public"
     case identity: AwsIdentity => s"${identity.stage.toLowerCase}.dev-gutools.co.uk.settings.public"
-    case development: DevIdentity => "local.dev-gutools.co.uk.settings.public"
+    case _: DevIdentity => "local.dev-gutools.co.uk.settings.public"
   }
   val publicSettings = new PublicSettings(settingsFile, "pan-domain-auth-settings", s3Client)
   publicSettings.start()
 
+  val typerighterBucket = identity match {
+    case identity: AwsIdentity => s"typerighter-${identity.stage.toLowerCase}"
+    case _: DevIdentity => "typerighter-code"
+  }
```

Comment:
Discussed briefly yesterday – if we use `typerighter` as the bucket name, there's fewer buckets at the root of the account, which is preferable as we have lots! I know there are a few `-code` examples in the Composer account, but I can't help but think that fewer buckets namespaced by domain == better – @akash1810 may also have an opinion here.

Reply:
That occurred to me too, the reason I made it specific was so that it would be unique to the stack. Would it cause issues if the CODE and PROD stack both wanted to create a bucket with the same name? OR would we need to create the buckets manually and not use CFN?

Reply:
Hm, you can adopt resources when creating a stack ... but I've never done it! Looking at the bucket list this makes sense – the buckets that do use this pattern tend to be `dist`, and uncoupled from application stacks.

There's not much in it and other apps use this pattern, so yeah, let's continue as is 👍

Reply:
I've not got strong opinions here, have seen both patterns being used. I do tend to favour a single bucket with stage directories inside it, but for no tangible reason.

With a single bucket, you'd trade the simplicity of a simple single bucket with the complexity of access policies as you'd likely want to restrict the CODE app to the CODE prefix in the bucket.

With multiple buckets, cost management becomes easier as you can add the stage tag to each bucket and then aggregate your costs using tags. This is only really useful if you're storing large files in the bucket though as the cost otherwise is negligible.

I don't have all the context so this might already have been considered, however, it looks like we're using S3 as a cache for a JSON blob? Would DynamoDB or RDS be a better fit for this?

> OR would we need to create the buckets manually and not use CFN?

Yes, you'd create it and make it a param to the stack.

Reply:
> Yes, you'd create it manually and make it a param to the stack.

Ah, gotcha! I had the tools to figure this out myself 🤦 

Re: S3 as a cache, S3 is effectively storing a readonly, 'published' version of these rules – right now, the rules in their editable state are stored in a google sheet, but eventually they'll be stored in an RDS or similar. So we use S3 as a high-availability place to put a published bundle of rules.

The 'refresh' action here is taken on the Typerighter matcher service right now, but eventually this will be the responsibility of an as-yet-hypothetical rule manager service, which will store rules in an editable state and 'publish' them on a user action – just in case  the current setup is a bit confusing!

V. happy to discuss further if anything above is unclear or doesn't sound like a good idea.

Reply:
Please never create anything manually. By all means have a common stack with shared resources if you really need to go that way. Personally I would argue that you should never share resource between stages - they should be completely isolated. It's slightly easier to reason about a CFN stack that doesn't share anything with different stages and permissions are also easier to reason about IMO.